Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -199,14 +202,12 @@ private static string GetSourceFileContent(string filePath, bool addNewLine, Enc
 
     }
 
-    /// <summary>
-    /// Uploads source file to destination. Overwrites the destination file if enabled.
-    /// </summary>
-    /// <param name="removeExisting"></param>
-    private void PutDestinationFile(bool removeExisting = false)
+    private void PutDestinationFile(CancellationToken cancellationToken, bool removeExisting = false)
     {
         var doRename = BatchContext.Options.RenameDestinationFileDuringTransfer;
 
+        cancellationToken.ThrowIfCancellationRequested();

```

Comment:
No need to check cancellation here.

Reply:
The Task has been known for getting stuck and cannot be terminated. This is the reason I added these cancellation checks in different layers of the Task.